tests/integration/lxd/test_launcher.py

```diff
@@ -575,7 +592,55 @@ def test_launch_instance_config_incompatible_with_auto_clean(core20_instance):
 
     core20_instance.push_file_io(
         destination=base_configuration.instance_config_path,
-        content=io.BytesIO(b"compatibility_tag: invalid\n"),
+        content=io.BytesIO(b"compatibility_tag: invalid\nsetup: true\n"),
+        file_mode="0644",
+    )
+
+    # when auto_clean is true, the instance will be deleted and recreated
+    lxd.launch(
+        name=core20_instance.name,
+        base_configuration=base_configuration,
+        image_name="20.04",
+        image_remote="ubuntu",
+        auto_clean=True,
```

Comment:
ah I think this answers my question regarding the auto-clean

Reply:
Bingo - the decision was to delete and recreate incomplete instances when `auto_clean=True`.